common/calibration-model.cpp

```diff
@@ -450,3 +524,324 @@ void calibration_model::update(ux_window& window, std::string& error_message)
+        }
+        if (ImGui::IsItemHovered())
+        {
+            window.link_hovered();
+            RsImGui::CustomTooltip("%s", "Close without saving any changes");
+        }
+        ImGui::SameLine();
+
+        if (_accept)
+        {
+            if (ImGui::Button(u8"\uF2DB  Write Table", ImVec2(120, 25)))
+            {
+                try
+                {
+                    auto actual_data = _calibration.data() + sizeof(librealsense::ds::table_header);
+                    auto actual_data_size = _calibration.size() - sizeof(librealsense::ds::table_header);
+                    auto crc = rsutils::number::calc_crc32( actual_data, actual_data_size );
+                    table->header.crc32 = crc;
+                    dev.as<rs2::auto_calibrated_device>().set_calibration_table(_calibration);
+                    dev.as<rs2::auto_calibrated_device>().write_calibration();
+                    _original = _calibration;
+                    orig_table = reinterpret_cast< librealsense::ds::d500_coefficients_table * >( _original.data() );
+                    ImGui::CloseCurrentPopup();
+                }
+                catch (const std::exception& ex)
+                {
+                    error_message = ex.what();
+                    ImGui::CloseCurrentPopup();
+                }
+            }
+            if (ImGui::IsItemHovered())
+            {
+                window.link_hovered();
+                RsImGui::CustomTooltip("%s", "Write selected calibration table to the device");
+            }
+        }
+        else
+        {
+            ImGui::PushStyleColor(ImGuiCol_Text, grey);
+            ImGui::PushStyleColor(ImGuiCol_TextSelectedBg, grey);
+
+            ImGui::Button(u8"\uF2DB  Write Table", ImVec2(120, 25));
+            if (ImGui::IsItemHovered())
+            {
+                RsImGui::CustomTooltip("%s", "Write selected calibration table to the device. For advanced users");
+            }
+
+            ImGui::PopStyleColor(2);
+        }
+        auto streams = dev.query_sensors()[0].get_active_streams();
+        if (changed)
+        {
+            try
+            {
+                dev.as<rs2::auto_calibrated_device>().set_calibration_table(_calibration);
+            }
+            catch (const std::exception&)
+            {
+                try
+                {
+                    dev.query_sensors()[0].close();
+                    dev.query_sensors()[0].open(streams);
+                    dev.as<rs2::auto_calibrated_device>().set_calibration_table(_calibration);
+                }
+                catch (const std::exception& ex)
+                {
+                    error_message = ex.what();
+                    ImGui::CloseCurrentPopup();
+                }
+            }
+        }
+
+        if (ImGui::IsWindowHovered()) window.set_hovered_over_input();
+
+        ImGui::EndPopup();
+    }
+    ImGui::PopStyleColor(3);
+    ImGui::PopStyleVar(2);
+}
```

Comment:
Check for a blank line at the end of file. Some compilers will fail if not there